src/core/js/a11y.js

```diff
@@ -0,0 +1,686 @@
+            this.config._isActive = false;
+            _.defaults(this.config, this.defaults);
+            Adapt.offlineStorage.set('a11y', false);
+            this.$html.toggleClass('accessibility', this.isEnabled());
+            this._setupNoSelect();
+            this._addFocuserDiv();
+            if (this._isReady) return;
+            this._isReady = true;
+            Adapt.trigger('accessibility:ready');
+        },
+
+        _setupNoSelect: function() {
+            if (!this.config || !this.config._disableTextSelectOnClasses) return;
+            var classes = this.config._disableTextSelectOnClasses.split(' ');
+            var isMatch = false;
+            for (var i = 0, item; item = classes[i++];) {
+                if (this.$html.is(item)) {
+                    isMatch = true;
+                    break;
+                }
+            }
+            this.$html.toggleClass('no-select', isMatch);
+        },
+
+        _addFocuserDiv: function() {
+            if ($('#a11y-focuser').length) return;
+            $('body').append($('<div id="a11y-focuser" class="a11y-ignore" tabindex="-1" role="presentation">&nbsp;</div>'));
+        },
+
+        _removeLegacyElements: function() {
+            var $legacyElements = $('body').children('#accessibility-toggle, #accessibility-instructions');
+            var $navigationElements = $('.navigation').find('#accessibility-toggle, #accessibility-instructions');
+            if (!$legacyElements.length && !$navigationElements.length) return;
+            Adapt.log.warn('REMOVED: #accessibility-toggle and #accessibility-instructions have been removed. Please remove them from all of your .html files.');
+            $legacyElements.remove();
+            $navigationElements.remove();
+        },
+
+        _onNavigationStart: function() {
+            if (!this.isEnabled()) return;
+            // Stop document reading
+            _.defer(function() {
+                Adapt.a11y.toggleHidden('.page, .menu', true);
+            });
+        },
+
+        _onNavigationEnd: function(view) {
+            // Prevent sub-menu items provoking behaviour
+            if (view && view.model && view.model.get('_id') !== Adapt.location._currentId) return;
+            if (!this.isEnabled()) return;
+            // Allow document to be read
+            Adapt.a11y.toggleHidden('.page, .menu', false);
+        },
+
+        isActive: function() {
+            this.log.removed('Accessibility is now always active when enabled. Please unify your user experiences.');
+            return false;
+        },
+
+        isEnabled: function() {
+            return this.config && this.config._isEnabled;
+        },
+
+        /**
+         * Adds or removes `aria-hidden` attribute to elements.
+         *
+         * @param {Object} $elements
+         * @param {boolean} [isHidden=true]
+         * @returns {Object} Returns `Adapt.a11y`
+         */
+        toggleHidden: function($elements, isHidden) {
```

Comment:
`$elements` is actually a JQuery selector.  It might make sense to rename this parameter and declare `var $element = $(selector);` below.

Reply:
Even though it _CAN_ be an selector, `$element = $($element)` is there to ensure that it is always converted into a jQuery object where it receives a string or Array or jQuery object [API](https://api.jquery.com/jQuery/), I'd prefer to leave it as is if that's ok?

Reply:
Good job we're not using Typescript.  :)

Yeah, that's ok.  Can you maybe update the JSDoc in that case though to indicate that the parameter can be any of those types?

Reply:
If it were typescript we'd just palm it off onto whatever madness the jQuery selector uses to define its attribute or return type, rather than just the generic "Object".

Reply:
https://github.com/andraaspar/jquery-ts/blob/master/src/IStatic.d.ts#L22

`string|any` ... how boring...